### DISCUSSION TITLE
Unable to publish actions to the GitHub Marketplace

### DISCUSSION THREAD

**Discussion by marvinpinto** [community/community#25844] [Actions]
<p>Hey there,</p>
<p>I am trying to publish two actions to the GitHub Marketplace and similarly to <a href="https://github.community/t/i-dont-see-publish-this-action-to-the-github-marketplace-when-creating-a-new-version/16391">this person</a>, I don’t see an option to “Publish this action to the GitHub Marketplace” according to the instructions <a href="https://developer.github.com/marketplace/actions/publishing-an-action-in-the-github-marketplace/" rel="nofollow noopener">here</a>. The actions that I would like to publish are in my <a href="https://github.com/marvinpinto/actions" rel="nofollow noopener">marvinpinto/actions</a> public repository.</p>
<p>Wondering if someone at GitHub can help me out with this?</p>

**Comment by cchantep** [comment]
<p>It seems to be a recurring issue, at least that means this topic is quite under-documented.</p>

**Comment by marvinpinto** [comment, marked as answer]
<p>I managed to sort this out myself. I suspected it was because my actions were in a monorepo and not in separate repos like most others, and validated that theory. I had to mirror the individual packages out to separate repos, but I got that done and my actions are now published in the marketplace.</p>
<p>Hope we get support for monorepos at some point!</p>

**Comment by cchantep** [comment]
<p>It also seems that even if the action is distributed on an orphan branch (or by tag based on such branch) where the metadata file <code>action.yml</code> is present, it won’t be detected as this file must be on <code>master</code> anyway.</p>
